### DISCUSSION TITLE
Why inserted text is not visible on some pdfs

### DISCUSSION THREAD

**Discussion by aarthihWN** [pymupdf/PyMuPDF#2152] [Looking for help] [answered]
Hello,
My current requirement is to paginate a bunch of pdfs sequentially with page number on the bottom right of the page. Below is the code I used, considering that many of the pages of the pdfs were rotated.  While this works for most pdfs, it does not for a few unusual files, with and without images. Have attached sample pdfs(or part of it). On one , the number does not show up at all. On the other, am not able to control the position. Please help. Thank you.
[SinglePage.pdf](https://github.com/pymupdf/PyMuPDF/files/10324843/SinglePage.pdf)
[15. R-012 Drawing Ref. 13139-SE-RES-0-003-02 1.pdf](https://github.com/pymupdf/PyMuPDF/files/10324845/15.R-012.Drawing.Ref.13139-SE-RES-0-003-02.1.pdf)

    page_number_for_print = 1
    for path in pathlist:
        doc = fitz.open(path)
        for page in doc:
            if page.rotation == 90:
                page.insert_text((page.rect.height-20,60),str(page_number_for_print), fontsize=28, rotate=90, color=(0, 0, 0))
            elif page.rotation == 180:
                page.insert_text((60, 20), str(page_number_for_print), fontsize=18, rotate=180,color=(0,0,0))
            elif page.rotation == 270:
                page.insert_text((20, page.rect.width - 60), str(page_number_for_print), stroke_opacity=3, fontsize=18,color=(0,0,0), rotate=270)                
            else:
                page.insert_text((page.rect.width - 60, page.rect.height-20), str(page_number_for_print), fontsize=18,color=(0,0,0))
            page_number_for_print += 1
        doc.saveIncr()
        doc.close()

**Comment by JorjMcKie** [comment, marked as answer]
Your only problem are "geometry changes":
If you do a `page.clean_contents()` before inserting anything, you should be fine.

BTW: The example files attached here are read-only, so you actually should get a "permission denied" or similar exception when using `"saveIncr()"`?

**Reply by aarthihWN** [reply]
Thank you! That worked.
Yes, these files are manually saved as editable before running the code. Is there a  way to check this in the code?

**Reply by JorjMcKie** [reply]
>  Is there a way to check this in the code?

not within PyMuPDF - yet. But of course no problem to do this check with the Python `os` module. The action taken in the problem case would be raising an exception however, which already is done today by the underlying C library.

**Reply by JorjMcKie** [reply]
In Python you can check whether a path / filename is writable via `if os.stat(filename).st_mode & 128: print("writable!")`.
Where `128 = stat.S_IWUSR`, a constant defined in the `stat` module.

**Reply by aarthihWN** [reply]
Thanks you. The file being read-only is not a problem as the files need to be saved along with other modifications as new files. So that is resolved. Thanks again for your help!
